sql/src/main/java/io/crate/action/sql/SQLOperations.java

```diff
@@ -240,18 +254,27 @@ public void parse(String statementName, String query, List<DataType> paramTypes)
             }
         }
 
-        public void bind(String portalName, String statementName, List<Object> params, @Nullable FormatCodes.FormatCode[] resultFormatCodes) {
+        public void bind(String portalName,
+                         String statementName,
+                         List<Object> params,
+                         @Nullable FormatCodes.FormatCode[] resultFormatCodes) {
             LOGGER.debug("method=bind portalName={} statementName={} params={}", portalName, statementName, params);
             checkError();
 
             try {
-                Portal portal = getOrCreatePortal(portalName, statementName);
-                portal.resultFormatCodes = resultFormatCodes;
-                portal.bulkParams.add(params);
-                if (portal.analysis == null) {
+                Portal portal = getOrCreatePortal(portalName, statementName, statement);
+
+                portal.addQuery(statementName, query);
```

Comment:
Could these `add` methods be changed to have a single `bind` method on the portal as well?
I think the stuff will only work if all these things are set, so the API should require all of them. If you have separate `add` methods it's not enforced and easier to use the API in a wrong way.